src/routes/getFiles.ts

```diff
@@ -0,0 +1,235 @@
+
+  const paths = Object.keys(body.paths)
+
+  // Use max page size config to limit the paths processed
+  if (paths.length > MAX_PAGE_SIZE) {
+    throw makeApiClientError(
+      422,
+      `Too many paths. MAX_PAGE_SIZE is ${MAX_PAGE_SIZE}`
+    )
+  }
+
+  const keys = paths.map(filePath => `${repoId}:${filePath}`)
+  // Maps a directory key to a file path for timestamp lookup
+  const filePathMapOfParentDirKey: { [dirKey: string]: string } = {}
+
+  // Fetch Paths:
+
+  const fetchPathsResults = await dataStore.fetch({ keys })
+
+  for (const row of fetchPathsResults.rows) {
+    const documentKey = row.key
+    // No need to check for undefined case here because of how keys
+    // are defined for the fetch.
+    const documentPath = documentKey.split(':')[1]
+
+    const requestTimestamp = body.paths[documentPath]
+
+    if ('doc' in row) {
+      let fileDocument: StoreFileDocument | undefined
+      let directoryDocument: StoreDirectoryDocument | undefined
+
+      try {
+        // For file document
+        fileDocument = asStoreFileDocument(row.doc)
+      } catch (err) {
+        try {
+          // For directory document
+          directoryDocument = asStoreDirectoryDocument(row.doc)
+        } catch (err) {
+          throw makeApiClientError(
+            422,
+            `File '${row.key}' is not a file or directory.`
+          )
+        }
+      }
+
+      // Handle file document
+      if (fileDocument !== undefined) {
+        responsePaths[documentPath] = {
+          ...asStoreFile(fileDocument),
+          timestamp: 0
+        }
+
+        // Add file path to file directory map for timestamp lookup
+        if (!ignoreTimestamps) {
+          const [parentDirectoryPath = '/'] = getParentPathsOfPath(documentPath)
+          const parentDirectoryKey = `${repoId}:${parentDirectoryPath}`
+          filePathMapOfParentDirKey[parentDirectoryKey] = documentPath
+        }
+      }
+
+      // Handle directory document
+      if (directoryDocument !== undefined) {
+        const storeDirectory = asStoreDirectory(directoryDocument)
+        const allTimestamps = [
+          ...Object.values(storeDirectory.deleted),
+          ...Object.values(storeDirectory.paths)
+        ]
+        const directoryTimestamp = Math.max(...allTimestamps)
+
+        if (ignoreTimestamps) {
+          const files = storeDirectory.paths
+          responsePaths[documentPath] = { files, timestamp: directoryTimestamp }
+          console.log('IGNORED FOR DIR')
+        } else if (directoryTimestamp > requestTimestamp) {
+          const files = storeDirectory.paths
+
+          console.log('NOT IGNORED FOR DIR')
+
+          // Remove files after timestamp check
+          Object.entries(storeDirectory.paths).forEach(
+            ([fileName, fileTimestamp]) => {
+              if (fileTimestamp <= requestTimestamp) {
+                delete files[fileName]
+              }
+            }
+          )
```

Comment:
@paullinator Currently, we're returning all files and directory names in the directory's paths property as the `files` property in the response. Should we be returning directories? If not, it'll require that we fetch all paths here to check whether they're file documents (which is not ideal). Alternatively, we could signify directories with a '/' suffix in the key string as was mentioned before which would make the file/directory distinction trivial, however it would require this semantic to be reinforced by `/updateFiles`.

Reply:
A directory's returned `files` should include sub-directories. Change here is to rename the `files` property to `paths` for clearer semantics.